File: text_ code/chap05/regexpr..py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct 28 13:05:55 2019

@author: Sim
"""

import re

mystr = re.sub('\\[', 'A',  '[12.5+13-1*5$')
print(mystr)

mystr = re.sub('[a-z]', '-',  'bcde0123ABC')
print(mystr)

l_case = re.search('[a-z]', 'bcde0123ABC')
if l_case: print('a-z exits')
else: print('No a-z')

l_case = re.search('[k-z]', 'bcde0123ABC')
if l_case: print('k-z exists')
else: print('No k-z')


mystr =  [re.sub('grey|gray', 'GRAY',  word) for word in 
     ['grey', 'gray', 'groy', 'greay', 'greey']]
print(mystr)

mystr=[re.sub('gr(a|e)y', 'GRAY', word) for word in
       ['grey', 'gray', 'groy', 'greay', 'greey']]
print(mystr)

File: text_ code/chap05/regexpr.r

gsub('\\[', 'A',  '[12.5+13-1*5$')
gsub('\\+', 'A',  '[12.5+13-1*5$')
gsub('\\$', 'A',  '[12.5+13-1*5$')


sub('[a-z]','-', 'bcde0123ABC')
gsub('[a-z]','-', 'bcde0123ABC')

sub('colou?r', 'COLOR', c('color', 'coloor', 'colooor', 'colour', 'colouur')) 
sub('colo*r', 'COLOR', c('color', 'coloor', 'colooor', 'colour', 'colouur')) 
sub('colo+r', 'COLOR', c('color', 'coloor', 'colooor', 'colour', 'colouur')) 
sub('colo{3}r', 'COLOR',  c('color', 'coloor', 'colooor', 'colour', 'colouur')) 
sub('colo{1,3}r', 'COLOR',  c('color', 'coloor', 'colooor', 'colour', 'colouur')) 
sub('colo{2,}r', 'COLOR',  c('color', 'coloor', 'colooor', 'colour', 'colouur')) 


sub('a.b', 'A.B',  c('ab', 'abb', 'abbb', 'axb', 'axxxxb')) 
sub('a.*b', 'A.B',  c('ab', 'abb', 'abbb', 'axb', 'axxxxb')) 

sub('grey|gray', 'GRAY',  c('grey', 'gray', 'groy', 'greay', 'greey')) 
sub('gr(a|e)y', 'GRAY',  c('grey', 'gray', 'groy', 'greay', 'greey'))   


sub('^1', '-', c('12.5', 'ab', 'abc', ''))
sub('[^0-9]', '-', c('12.5', 'ab', 'abc', ''))
sub('b$', '-', c('12.5', 'ab', 'abc', ''))
sub('^$', '-', c('12.5', 'ab', 'abc', ''))
